provisioning/service/src/Config/QueryResult.cs

```diff
@@ -0,0 +1,173 @@
+    ///         service will return the next page of this query when you send a new query with
+    ///         this token.</item>
+    /// </list>
+    ///
+    /// And the body is a JSON list of the specific <b>type</b>. For instance, if the system
+    ///     property type is IndividualEnrollment, the body will look like:
+    /// <code>
+    /// [
+    ///     {
+    ///         "registrationId":"validRegistrationId-1",
+    ///         "deviceId":"ContosoDevice-1",
+    ///         "attestation":{
+    ///             "type":"tpm",
+    ///             "tpm":{
+    ///                 "endorsementKey":"validEndorsementKey"
+    ///             }
+    ///         },
+    ///         "iotHubHostName":"ContosoIoTHub.azure-devices.net",
+    ///         "provisioningStatus":"enabled"
+    ///     },
+    ///     {
+    ///         "registrationId":"validRegistrationId-2",
+    ///         "deviceId":"ContosoDevice-2",
+    ///         "attestation":{
+    ///             "type":"tpm",
+    ///            "tpm":{
+    ///                 "endorsementKey":"validEndorsementKey"
+    ///             }
+    ///         },
+    ///         "iotHubHostName":"ContosoIoTHub.azure-devices.net",
+    ///         "provisioningStatus":"enabled"
+    ///     }
+    /// ]
+    /// </code>
+    /// </remarks>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/deviceenrollment">Device Enrollment</seealso>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/deviceenrollmentgroup">Device Enrollment Group</seealso>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/registrationstatus">Registration Status</seealso>
+    public class QueryResult
+    {
+        /// <summary>
+        /// Getter for the query result Type.
+        /// </summary>
+        public QueryResultType Type { get; private set; }
+
+        /// <summary>
+        /// Getter for the list of query result Items.
+        /// </summary>
+        public IEnumerable<Object> Items { get; private set; }
+
+        /// <summary>
+        /// Getter for the query result continuationToken.
+        /// </summary>
+        public string ContinuationToken { get; private set; }
+
+        /// <summary>
+        /// CONSTRUCTOR
+        /// </summary>
+        /// <param name="typeString">the <code>string</code> with type of the content in the body. It cannot be <code>null</code>.</param>
+        /// <param name="bodyString">the <code>string</code> with the body in a JSON list format. It cannot be <code>null</code>, or empty, if the type is different than `unknown`.</param>
+        /// <param name="continuationToken">the <code>string</code> with the continuation token. It can be <code>null</code>.</param>
+        /// <exception cref="ArgumentException">if one of the provided parameters is invalid.</exception>
+        internal QueryResult(string typeString, string bodyString, string continuationToken)
```

Comment:
This should be public: developers need a way to construct types like these for their own unit testing.

Reply:
As a general ruler, we do not expose unnecessary methods, this constructor was made to fill this object with the server result. Make it public will create a unnecessary public API.

Reply:
I think it's OK not to expose this: users can still wrap this in their own interface and use mock that instead.